syncopy/shared/parsers.py

```diff
@@ -340,9 +340,17 @@ def array_parser(var, varname="", ntype=None, hasinf=None, hasnan=None,
     scalar_parser : similar functionality for parsing numeric scalars
     """
 
-    # Make sure `var` is array-like and convert it to ndarray to simplify parsing
+    # Make sure `var` is array-like
     if not isinstance(var, (np.ndarray, list)):
         raise SPYTypeError(var, varname=varname, expected="array_like")
+
+    # "Exotic" arrays (str et al.) must contain only elements of the same type
```

Comment:
Ok this works indeed nicely now, and also makes the `sequence_parser` for function arguments like `('ch1', 'ch2')` not obsolete, as the array parser really expects arrays, which is good!